src/magic_folder/test/test_tahoe_client.py

```diff
@@ -248,3 +252,46 @@ def test_mutable_directories_distinct(self, children):
                 ),
             ),
         )
+
+    @given(
+        just("foo"), #path_segments(),
+        just(b"test"*200), #binary(),
+    )
+    def test_mutable_directory_add_child(self, child_name, content):
+        """
+        A child can be added to a mutable directory
+        """
+        mutable_d = self.tahoe_client.create_mutable_directory()
+        child_d = self.tahoe_client.create_immutable(content)
+        self.assertThat(gatherResults([mutable_d, child_d]), Always())
+
+        mutable_cap = mutable_d.result
+        child_cap = child_d.result
+
+        self.tahoe_client.add_entry_to_mutable_directory(
+            mutable_cap,
+            child_name,
+            child_cap,
+        )
+
+        child_uri = ANY_ROOT.child(u"uri", mutable_cap.decode("utf8"), child_name)
+        resp_d = self.http_client.get(child_uri.to_text())
+        self.assertThat(
+            succeeded(resp_d),
```

Comment:
I have not understood how this test works.  Does `succeeded(resp_d)` mean that a success response is expected here?  How can that be, since `ANY_ROOT` an invalid URL?

Reply:
Good catch, this is the wrong way to use the API. It should be `assertThat(resp_d, succeeded(Always()))`

As for the URL what is essentially happening is that only the "path" part is being used; the Agent etc down into `treq` makes it not examine the host part of the URL for the TCP connection.